pyproject.toml

```diff
@@ -26,6 +26,8 @@ exclude = [
 ]
 
 [project.optional-dependencies]
+cudf = ["cudf>=23.08.00"]
+modin = ["modin"]
```

Comment:
not sure what the minimum should be yet, but i've opened https://github.com/narwhals-dev/narwhals/issues/817 to track this